espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/InputSource.java

```diff
@@ -0,0 +1,190 @@
+/**
+ * InputSource
+ *
+ * (refer to https://www.w3.org/TR/webdriver/#terminology-0 of W3C spec)
+ *
+ * Represents a Virtual Device providing input events
+ */
+package io.appium.espressoserver.lib.helpers.w3c.models;
+
+import com.google.gson.annotations.SerializedName;
+
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+@SuppressWarnings("unused")
+public class InputSource {
+    private final String VIEWPORT = "viewport";
+    private final String POINTER = "pointer";
+
+    private InputSourceType type;
+    private String id;
+    private Parameters parameters;
+    private List<Action> actions;
+
+    public String getId() {
+        return id;
+    }
+
+    public List<Action> getActions() {
+        return actions;
+    }
+
+    public void setActions(List<Action> actions) {
+        this.actions = actions;
+    }
+
+    public InputSourceType getType() {
+        if (type == null) {
+            return InputSourceType.NONE;
+        }
+        return type;
+    }
+
+    public void setType(InputSourceType type) {
+        this.type = type;
+    }
+
+    @Nullable
+    public PointerType getPointerType(){
+        if(parameters != null) {
+            return parameters.getPointerType();
+        }
+        return null;
```

Comment:
I think we can make it TOUCH by default

Reply:
That's coming in the next PR.... and we make it TOUCH by default, only if it's type POINTER